light_controller/ir_toy/ir_toy.py

```diff
@@ -0,0 +1,351 @@
+
+    def setMuteLED_Off(self):
+        """
+        Enables the LED during sample mode
+        :return: None
+        """
+        self.writeByte(CommandEnum.setLED_MuteOff)
+
+    def setLED_On(self):
+        """
+        Turn the LED on
+        :return: None
+        """
+        self.writeByte(CommandEnum.setLED_On)
+
+    def setLED_Off(self):
+        """
+        Turn the LED off
+        :return: None
+        """
+        self.writeByte(CommandEnum.setLED_Off)
+
+    def getSettings(self):
+        """
+        Returns 8 bytes that describe the current IR Toy settings:
+
+        Descriptor fields
+        Bytes	Description
+        1		PR2 value (PWM)
+        1		Duty cycle
+        1		PWM prescaler, 2 extra bits of duty cycle
+        1		Transmit timer prescaler
+        4		Clock frequency in Hz (eg 48,000,000Hz)
+        :return: Descriptor Fields
+        """
+        self.writeByte(CommandEnum.getSettingsReport)
+        settingsBuffer = self.read(8)
+        print(len(settingsBuffer))
+        report = IR_ToySettings(*struct.unpack(">BBBBL", settingsBuffer))
+        return report
+
+    def enableTX_ByteCount(self):
+        """
+        If enabled, the IR Toy returns the number of bytes transmitted at the end of a 0x03 command. The format is tXY,
+        where XY is the 16bit count of bytes transmitted:
+
+        't' is the character t
+        X is the high 8 bits of byte count
+        Y is the low 8 bits of byte count
+        :return: None
+        """
+        self.writeByte(CommandEnum.enableTX_ByteReport)
+
+    def enableTX_NotifyOnComplete(self):
+        """
+        If enabled, the IR Toy returns a success or error code at the end of a 0x03 transmit command.
+
+        Notify on complete codes
+        Code	Description
+        C		Success, no errors, complete
+        F		Buffer underrun during transmit
+        :return: None
+        """
+        self.writeByte(CommandEnum.enableTX_NotifyOnComplete)
+
+    def enableTX_Handshake(self):
+        """
+        If enabled, the IR Toy will request data packets during a transmit command. Each request is the number of bytes
+        free in the USB buffer, currently always 62. The software interfacing the IR Toy should wait for each request
+        before sending the data.
+        :return: None
+        """
+        self.writeByte(CommandEnum.enableTX_Handshake)
+
+    def reset(self):
+        """
+        Reset the device
+        :return: None
+        """
+        self.writeArray(self.TERMINATOR)
+        self.writeArray((CommandEnum.reset,) * 5)
+        time.sleep(0.05)
+
+    def getVersion(self):
+        """
+        Get the hardware and firmware versions
+        :return: hardware string, firmware version
+        """
+        try:
+            self.reset()
+            self.writeByte(ord('v'))
+            buffer = self.read(4)
+            hardware, revision = struct.unpack('>2s2s', buffer)
+            return hardware, int(revision)
```

Comment:
add an exception clause, to perhaps emit the reason for failure for debugging.  Also some general retry logic.